src/alt.js

```diff
@@ -34,6 +34,13 @@ function uid(container, name) {
   return key
 }
 
+/* for now just detect whether it's instance of Immutable.js */
+function isImmutable(state) {
+  // let's check with that for now
+  // https://github.com/facebook/immutable-js/issues/421
+  return typeof state.toJS === 'function'
```

Comment:
FWIW I don't mind supporting immutable-js as first class into alt, it has a lot of traction and there's been enough people asking for it. The trick is to do it in a minimal and elegant way.